wagtail/admin/checks.py

```diff
@@ -166,6 +166,72 @@ def check_panels_in_model(cls, context="model"):
     return errors
 
 
+@register("panels")
+def panel_type_check(app_configs, **kwargs):
```

Comment:
Sorry @lb- before we get this in, I haven't really reviewed this but just wanted to make sure, have you looked into how this relates to this existing check? e.g. whether this could be merged into that check, or whether the traversal is something we'd want to do only once for all checks that require traversal, etc.

https://github.com/wagtail/wagtail/blob/4a097466ebfa3260917c66f8a1e6f2891ebae399/wagtail/admin/checks.py#L105

Reply:
The existing check is about the case where we have `edit_handler = ...`  vs using `content_panels/other_panels = `. Making sure that these are being used correctly as there is a subtly different behaviour between Pages and non-Page models.

It's not really checking the same thing, this new check is for the inner individual panel items mapping to a correct field.

Happy to hold on merging though if you wanted to do a review.